lips/lip-6.md

```diff
@@ -0,0 +1,88 @@
+
+# Transaction Scripts
+The following transaction scripts can be run by the LibraRoot:
+* `create_validator_account`
+* `create_validator_operator_account`
+* `add_validator_and_reconfigure`
+* `remove_validator_and_reconfigure`
+
+The following transaction scripts can be run by the validator owner:
+* `set_validator_operator`
+* `remove_validator_operator`
+* `rotate_authentication_key`
+
+The following transaction scripts can be run by the validator operator:
+* `register_validator_config`
+* `set_validator_config_and_reconfigure`
+* `rotate_authentication_key`
+
+# Notes
+In first version of Libra Protocol, validator owners will not maintain ownership of their Libra accounts, the LibraRoot account will run validator owners' operations on their behalf. Validator operators will maintain full ownership of their Libra accounts.
+Gradually, the validator owners will start managing their accounts themselves. They will send their new account key to the Association and the Association will rotate their account key to this new key. Right after this rotation transaction gets executed, the validator owner starts to maintain its Libra account.
+
+Both the validator owner account and the validator operator account store a human_name which refers to the name of the company behind the account written in snake case (lowercase, where spaces are replaced with underscores), it is initialized at account creation and never changes. Most validator-related transaction scripts get the address and the names of the account and check that the human_name under the account matches the name passed into the transaction script. This prevents accounts from accidentally mix up the addresses, which are hex strings, and adds a second level of protection against human errors.
+
+### Formal verification guarantees on the Move Implementation
+
+* Disciplined aborts
+* Only LibraRoot or a Validator Operator can trigger reconfiguration
+* Every change to an validator set triggers a reconfiguration
+* Only a validator can set its operator
+* Only an operator can update an on-chain config
+* The size of the validator set can only change by at most one when LibraConfig::NewEpochEvent is emitted
+* Multiple calls to add_validator or remove_validator are prohibited except in genesis
+* Both functions register_validator_config or set_validator_config_and_reconfigure check the validity of the consensus key and abort if an invalid key is given
+* An account can only appear once in a validator set
+* (currently) If the validator's operator is not set, the validator owner itself acts as its validator operator
+* If the validator owner changes the operator, its old config continues to persist in the validator set
```

Comment:
perhaps replace "continues to persist" with "does not change until explicit remove/add actions on the validator set are triggered.